apps/whale-api/src/module.api/governance.controller.ts

```diff
@@ -0,0 +1,64 @@
+import { ListProposalsStatus, ListProposalsType } from '@defichain/jellyfish-api-core/dist/category/governance'
+import {
+  GovernanceProposal,
+  ProposalMasternodeType,
+  ProposalVotesResult
+} from '@defichain/whale-api-client/dist/api/governance'
+import { Controller, Get, Param, ParseIntPipe, Query } from '@nestjs/common'
+import { GovernanceService } from './governance.service'
+import { ApiPagedResponse } from './_core/api.paged.response'
+import { PaginationQuery } from './_core/api.query'
+
+@Controller('/governance')
+export class GovernanceController {
+  constructor (private readonly governanceService: GovernanceService) {}
+
+  /**
+   * Return paginated governance proposals.
+   *
+   * @param {ListProposalsStatus} [status=ListProposalsStatus.ALL] type of proposals
+   * @param {ListProposalsType} [type=ListProposalsType.ALL] status of proposals
+   * @param {number} [cycle=0]  cycle: 0 (show all), cycle: N (show cycle N), cycle: -1 (show previous cycle)
+   * @param {PaginationQuery} query pagination query
+   * @returns {Promise<ApiPagedResponse<GovernanceProposal>>}
+   */
+  @Get('/proposals')
+  async listProposals (
+    @Query() status: ListProposalsStatus = ListProposalsStatus.ALL,
+    @Query() type: ListProposalsType = ListProposalsType.ALL,
```

Comment:
Not sure if we want to implement any enum-based validation here

Since it's an enum in jellyfish, it's likely that we're expected to manually keep this in sync with `ain`?
Or is this some Whale-specific enum?

Reply:
added into `governance.test.ts` using nestjs built-in pipe at the query string level